lib/ansible/modules/cloud/kubevirt/kubevirt_vm.py

```diff
@@ -247,92 +252,118 @@ def argspec(self):
-        definition = {'metadata': {'name': self.name, 'namespace': self.namespace}, 'spec': {'running': running}}
-        self.patch_resource(resource, definition, existing, self.name, self.namespace, merge_type='merge')
-
-        if wait:
-            resource = self.find_supported_resource('VirtualMachineInstance')
-            w, stream = self._create_stream(resource, self.namespace, wait_timeout)
-
-        if wait and stream is not None:
-            self._read_stream(resource, w, stream, self.name, running)
-
-    def _read_stream(self, resource, watcher, stream, name, running):
-        """ Wait for ready_replicas to equal the requested number of replicas. """
-        for event in stream:
-            if event.get('object'):
-                obj = ResourceInstance(resource, event['object'])
-                if running:
-                    if obj.metadata.name == name and hasattr(obj, 'status'):
-                        phase = getattr(obj.status, 'phase', None)
-                        if phase:
-                            if phase == 'Running' and running:
-                                watcher.stop()
-                                return
-                else:
-                    # TODO: wait for stopped state:
-                    watcher.stop()
-                    return
-
-        self.fail_json(msg="Error waiting for virtual machine. Try a higher wait_timeout value. %s" % obj.to_dict())
-
-    def manage_state(self, state):
-        wait = self.params.get('wait')
-        wait_timeout = self.params.get('wait_timeout')
-        resource_version = self.params.get('resource_version')
-
-        resource_vm = self.find_supported_resource('VirtualMachine')
-        existing = self.get_resource(resource_vm)
-        if resource_version and resource_version != existing.metadata.resourceVersion:
-            return False
-
-        existing_running = False
-        resource_vmi = self.find_supported_resource('VirtualMachineInstance')
-        existing_running_vmi = self.get_resource(resource_vmi)
-        if existing_running_vmi and hasattr(existing_running_vmi.status, 'phase'):
-            existing_running = existing_running_vmi.status.phase == 'Running'
-
-        if state == 'running':
-            if existing_running:
-                return False
+    @staticmethod
+    def fix_serialization(obj):
+        if obj and hasattr(obj, 'to_dict'):
+            return obj.to_dict()
+        return obj
+
+    def _wait_for_vmi_running(self):
+        for event in self._kind_resource.watch(namespace=self.namespace, timeout=self.params.get('wait_timeout')):
+            entity = event['object']
+            if entity.metadata.name != self.name:
+                continue
+            status = entity.get('status', {})
+            phase = status.get('phase', None)
+            if phase == 'Running':
+                return entity
+
+        self.fail("Error waiting for virtual machine. Maybe try a higher wait_timeout value?")
```

Comment:
What about `Timeout occurred while waiting on virtual machine to start`?

Reply:
done